### PR TITLE
feature/AD-1196_enable_using_authorized_user_token

### PR DESCRIPTION
**Pull request by LokmaneKrizou**

### CODE REVIEW

**Review by sjulia7 — APPROVED** [review]

**Review by Fabio-Insolia-LiveStyled — APPROVED** [review]

**Comment by Fabio-Insolia-LiveStyled** [comment]
@LokmaneKrizou once this is merged can we try to have a clean cut and update master?

File: core-sdk/src/main/java/com/realifetech/core_sdk/domain/CoreConfiguration.kt [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.realifetech.core_sdk.domain
 
+import android.annotation.SuppressLint
 import android.content.Context
 
+@SuppressLint("StaticFieldLeak")
```
[the remark this answers is not in the document]
Comment (LokmaneKrizou) [reply core-sdk/src/main/java/com/realifetech/core_sdk/domain/CoreConfiguration.kt:6]: Well, i saw the whole file of CoreConfiguration is complaining about the leak, the reason is we are saving context in this object, and we shouldn't do that, for now i added this so we can spot it later when we want to organise the code better.

File: core-sdk/src/main/java/com/realifetech/core_sdk/data/auth/RTLToken.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.realifetech.core_sdk.data.auth
+
+import com.google.gson.annotations.SerializedName
+import java.util.*
+
+data class RTLToken(
```
[the remark this answers is not in the document]
Comment (LokmaneKrizou) [reply core-sdk/src/main/java/com/realifetech/core_sdk/data/auth/RTLToken.kt]: we are rebranding to ReaTechLife didn't you hear about that XD

File: core-sdk/src/main/java/com/realifetech/core_sdk/network/OAuth2Authenticator.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.realifetech.core_sdk.network
+
+import com.realifetech.core_sdk.data.shared.`object`.toBearerFormat
+import com.realifetech.core_sdk.database.preferences.Preferences
+import okhttp3.Authenticator
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.Route
+
+class OAuth2Authenticator : Authenticator {
```
[the remark this answers is not in the document]
Comment (LokmaneKrizou) [reply core-sdk/src/main/java/com/realifetech/core_sdk/network/OAuth2Authenticator.kt]: i have added it now, we are refreshing the token using this method: accessToken.ensureActive()

**Pull request merged**